### PR TITLE
feat: Open source public API

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- IMPORTANT! Do not squash merge this PR in order to maintain repo history!

- Makes our API code open source.

## TODO before merging

- [ ] Update Vercel API deploy configs

### CODE REVIEW

File: apps/api/apps/api/jest.setup.ts
Comment (zomars) [inline apps/api/apps/api/jest.setup.ts:1]: cc @leog maybe we remove this with the Vitest migration

File: apps/api/apps/api/lib/validations/shared/timeZone.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import tzdata from "tzdata";
```
[the remark this answers is not in the document]
Comment (zomars) [reply apps/api/apps/api/lib/validations/shared/timeZone.ts:1]: Yes it was a workaround for deploying with the Vercel-deploy.sh script. Will cleanup in a followup.

File: apps/api/apps/api/LICENSE [hunk 2]
```diff
@@ -0,0 +1,42 @@
+The Cal.com Enterprise Edition (EE) license (the “EE License”)
```
[the remark this answers is not in the document]
Comment (PeerRich) [reply apps/api/apps/api/LICENSE:1]: oh right. lets launch with EE and then change license after!

File: apps/api/apps/api/README.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  <a href="https://cal.com/enterprise">Get Started with Enterprise</a>
+</div>
+
+# Cal.com Public API - Enterprise Edition
+
+Welcome to the Public API Enterprise Edition ("/apps/api") of the Cal.com Public API.
+
+The [/apps/api](https://github.com/calcom/cal.com/tree/main/apps/api) subfolder is the place for our Public API, which we serve at api.cal.com, and enterprise customers can also run [enterprise-grade](https://cal.com/enterprise)
+
+> _❗ WARNING: This repository is copyrighted (unlike our [main repo](https://github.com/calcom/cal.com)). You are not allowed to use this code to host your own version of app.cal.com without obtaining a proper [license](https://cal.com/pricing?infra) first❗_
```
Comment (PeerRich) [inline apps/api/apps/api/README.md:16]:
```suggestion
```

File: apps/api/apps/api/README.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
+<!-- PROJECT LOGO -->
+<div align="center">
+  <a href="https://cal.com/enterprise">
+    <img src="https://user-images.githubusercontent.com/8019099/133430653-24422d2a-3c8d-4052-9ad6-0580597151ee.png" alt="Logo">
```
Comment (PeerRich) [inline apps/api/apps/api/README.md:4]:
```suggestion
  <a href="https://cal.com/docs/enterprise-features/api#api-server-specifications">
    <img src="https://user-images.githubusercontent.com/8019099/133430653-24422d2a-3c8d-4052-9ad6-0580597151ee.png" alt="Logo">
```

File: apps/api/apps/api/README.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  </a>
+  
+  <a href="https://cal.com/enterprise">Get Started with Enterprise</a>
```
Comment (PeerRich) [inline apps/api/apps/api/README.md:7]:
```suggestion
  <a href="https://user-images.githubusercontent.com/8019099/133430653-24422d2a-3c8d-4052-9ad6-0580597151ee.png">Read the API docs</a>
```

File: apps/api/apps/api/README.md (hunk 3 above)
Comment (PeerRich) [inline apps/api/apps/api/README.md:16]:
```suggestion
# Cal.com Public API 

Welcome to the Public API ("/apps/api") of the Cal.com.
```

File: apps/api/apps/api/README.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+## Hosted api through cal.com
+
+> _❗ WARNING: This is still experimental and not fully implemented yet❗_
+
+Go to console.cal.com
+Add a deployment or go to an existing one.
+Activate API or Admin addon
+Provide your `DATABASE_URL`
+Now you can call api.cal.com?key=CALCOM_LICENSE_KEY, which will connect to your own databaseUrl.
+
```
Comment (PeerRich) [inline apps/api/apps/api/README.md:202]: is this still accurate after merging?

**Comment by PeerRich** [comment]
@baileypumfleet made a point, is it worth merging this vs rewriting the API (as planned) and keeping the legacy API closed source?

**Comment by zomars** [comment]
> @baileypumfleet made a point, is it worth merging this vs rewriting the API (as planned) and keeping the legacy API closed source?

Whatever the decision I think we should publish now even for historical purposes. Even if it's not merged the code is public now with this PR. So it will be more effort to rewrite history. here.

**Comment by zomars** [comment]
Merged after final approval from @baileypumfleet 🙏🏽

**Pull request merged**